aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbCatalog.java

```diff
@@ -248,7 +248,7 @@ public boolean dropNamespace(Namespace namespace) throws NamespaceNotEmptyExcept
       dynamo.deleteItem(DeleteItemRequest.builder()
           .tableName(awsProperties.dynamoDbTableName())
           .key(namespacePrimaryKey(namespace))
-          .conditionExpression("attribute_exists(" + namespace + ")")
+          .conditionExpression("attribute_exists(" + DynamoDbCatalog.COL_NAMESPACE + ")")
```

Comment:
Is this `conditionExpression` even needed? Are there any records that don't have a key `namespace` in them?

I guess if users are adding their own metadata to records or have a projected view (GSI)?

Reply:
Good question -- I'm not 100% sure, there may be edge cases that I'm not aware of.

I think that removing the `conditionExpression` might involve tweaking the exception handling [below](https://github.com/apache/iceberg/pull/3035/files#diff-76f1ec8211667ffbb03d3c237f7edff00812ccffb5b71b7d000989526e1bebd9R254) as well, I wouldn't expect the `ConditionalCheckFailedException` to be thrown any more.

Reply:
yes you are right, it was used to prevent calling the same method twice, but it likely does not matter given it's dropping the namespace. Thanks for noticing this bug, somehow I did not have a drop namesapce test. I will add it in another PR.

Reply:
nit: `COL_NAMESPACE` instead of `DynamoDbCatalog.COL_NAMESPACE`